### PR TITLE
Add internal error code to NetworkError and change draining error code

### PR DESCRIPTION
**Pull request by yunho-microsoft**
Add a new property: internalErrorCode to NetworkError class.
It is for giving more error information beside the code property.
Also change the draining error to include this property.

### CODE REVIEW

**Review by shubhi1092 — APPROVED** [review]
Looks good to me. Just wondering why the pnpm-lock changes are needed?

File: pnpm-lock.yaml
Comment (znewton) [inline pnpm-lock.yaml:1]: This shouldn't be changed by this PR

**Review by znewton — CHANGES_REQUESTED** [review]

File: pnpm-lock.yaml
Comment (yunho-microsoft) [reply pnpm-lock.yaml:1]: Good catch. Not sure why this file get changed. Revert it back

File: server/routerlicious/packages/lambdas/src/nexus/connect.ts [hunk 1]
```diff
@@ -334,7 +336,11 @@ async function checkClusterDraining(
 			...properties,
 			tenantId: message.tenantId,
 		});
-		throw new NetworkError(503, "Cluster is not available. Please retry later.");
+		const error = createFluidServiceNetworkError(503, {
+			message: "Cluster is not available. Please retry later.",
```
Comment (znewton) [inline server/routerlicious/packages/lambdas/src/nexus/connect.ts:340]: Message says "retry later", but there is no `retryAfter` value. Should there be?

**Review by znewton — APPROVED** [review]
Looks good, one comment

**Pull request merged**